---
# Tasks for for testing feature to solve Issue 436

- name: Insert a record into a collection
  community.mongodb.mongodb_shell:
    login_user: '{{ mongodb_admin_user }}'
    login_password: '{{ mongodb_admin_password }}'
    login_port: 3001
    debug: yes
    db: "test"
    eval: 'db.hosts.insert({"_id": "first_record"})'
    mongo_cmd: "{{ mongo_cmd }}"
  register: first_insert

- assert:
    that: 
      - first_insert.changed

- name: Insert a duplicate record into a collection - mongo shell does not throw an error so neither do we (mongosh does throw an error)
  community.mongodb.mongodb_shell:
    login_user: '{{ mongodb_admin_user }}'
    login_password: '{{ mongodb_admin_password }}'
    login_port: 3001
    debug: yes
    db: "test"
    eval: 'db.hosts.insert({"_id": "first_record"})'
    mongo_cmd: "{{ mongo_cmd }}"
    ignore_errors: "{{ 'yes' if mongodb_version.startswith('5') else 'no' }}"
  register: first_insert_duplicate

- assert:
    that: 
      - first_insert_duplicate.changed


- debug:
    var: first_insert_duplicate
